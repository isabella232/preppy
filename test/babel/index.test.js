import pify from "pify"
import { exec } from "child_process"
import { readFile } from "fs"
import rimraf from "rimraf"

import pkg from "../../package.json"

const versionString = `preppy v${pkg.version}`

const lazyExec = pify(exec)
const lazyRead = pify(readFile)
const lazyDelete = pify(rimraf)

process.chdir(__dirname)

jest.setTimeout(20000)

test("Publish Test File via Babel", async () => {
  await lazyDelete("./dist")

  console.log(await lazyExec(
    "node ../../bin/preppy --input-lib ./index.js --output-folder ./dist"
  ))

  const cjs = await lazyRead("./dist/index.cjs.js", "utf8")
  expect(cjs.replace(versionString, "VERSION_STRING")).toMatchSnapshot()

  const esm = await lazyRead("./dist/index.esm.js", "utf8")
  expect(esm.replace(versionString, "VERSION_STRING")).toMatchSnapshot()

  const umd = await lazyRead("./dist/index.umd.js", "utf8")
  expect(umd.replace(versionString, "VERSION_STRING")).toMatchSnapshot()
})
