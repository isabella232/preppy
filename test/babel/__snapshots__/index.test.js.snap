// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Publish Test File via Babel 1`] = `
"/*! VERSION_STRING by Sebastian Werner <s.werner@sebastian-software.de> */
'use strict';

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var _inheritsLoose = _interopDefault(require('@babel/runtime/helpers/inheritsLoose'));
require('core-js/modules/es6.object.assign');
require('core-js/modules/web.dom.iterable');
require('core-js/modules/es6.array.iterator');
require('core-js/modules/es6.string.iterator');
require('core-js/modules/es6.set');
require('core-js/modules/es7.array.includes');
require('core-js/modules/es6.promise');
var lodash = require('lodash');

console.log(\\"CherryPick Import Lodash:\\", lodash.camelCase(\\"hello world\\") === \\"helloWorld\\");
console.log(\\"Package\\", \\"preppy\\");
console.log(\\"Target\\", \\"lib\\");
console.log(\\"ES2016 Enabled:\\", Math.pow(2, 2) === 4);
new Promise(function (resolve, reject) {
  resolve(\\"resolved\\");
}).then(function (first) {
  console.log(\\"Promise:\\", first);
});
var CONSTANT = 123;
console.log(\\"Constant:\\", CONSTANT);
var myArray = [1, 2, 3];
console.log(\\"Supports Array.includes?:\\", myArray.includes && myArray.includes(2));
var mySet = new Set(myArray);
console.log(\\"Supports Set:\\", mySet.add(4));

(function (supportsDefault) {
  if (supportsDefault === void 0) {
    supportsDefault = true;
  }

  console.log(\\"Supports default parameters:\\", supportsDefault);
})();
/* eslint-disable no-shadow */


var testVariable = \\"outer\\";
var _testVariable = \\"inner\\";
console.log(\\"X Value from inner scope:\\", _testVariable);
console.log(\\"X Value from outer scope:\\", testVariable);
var source = {
  first: 1,
  second: 2
};
var destructed = Object.assign({
  third: 3
}, source);
console.log(\\"Destructed:\\", destructed);

var MyClass =
/*#__PURE__*/
function () {
  function MyClass() {
    console.log(\\"Called constructor\\");
    this.helper();
    this.instanceProperty = 3;
  }

  var _proto = MyClass.prototype;

  _proto.helper = function helper() {
    console.log(\\"Called helper\\");
  };

  return MyClass;
}();

var SecondClass =
/*#__PURE__*/
function (_MyClass) {
  _inheritsLoose(SecondClass, _MyClass);

  function SecondClass() {
    return _MyClass.call(this, 100) || this;
  }

  return SecondClass;
}(MyClass());

console.log(\\"Initialized class:\\", new MyClass());

function helper() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(PromiseHelper()).then(function ($await_1) {
      try {
        return $return(42);
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}

helper();
//# sourceMappingURL=index.cjs.js.map
"
`;

exports[`Publish Test File via Babel 2`] = `
"/*! VERSION_STRING by Sebastian Werner <s.werner@sebastian-software.de> */
import _inheritsLoose from '@babel/runtime/helpers/inheritsLoose';
import 'core-js/modules/es6.object.assign';
import 'core-js/modules/web.dom.iterable';
import 'core-js/modules/es6.array.iterator';
import 'core-js/modules/es6.string.iterator';
import 'core-js/modules/es6.set';
import 'core-js/modules/es7.array.includes';
import 'core-js/modules/es6.promise';
import { camelCase } from 'lodash';

console.log(\\"CherryPick Import Lodash:\\", camelCase(\\"hello world\\") === \\"helloWorld\\");
console.log(\\"Package\\", \\"preppy\\");
console.log(\\"Target\\", \\"lib\\");
console.log(\\"ES2016 Enabled:\\", Math.pow(2, 2) === 4);
new Promise(function (resolve, reject) {
  resolve(\\"resolved\\");
}).then(function (first) {
  console.log(\\"Promise:\\", first);
});
var CONSTANT = 123;
console.log(\\"Constant:\\", CONSTANT);
var myArray = [1, 2, 3];
console.log(\\"Supports Array.includes?:\\", myArray.includes && myArray.includes(2));
var mySet = new Set(myArray);
console.log(\\"Supports Set:\\", mySet.add(4));

(function (supportsDefault) {
  if (supportsDefault === void 0) {
    supportsDefault = true;
  }

  console.log(\\"Supports default parameters:\\", supportsDefault);
})();
/* eslint-disable no-shadow */


var testVariable = \\"outer\\";
var _testVariable = \\"inner\\";
console.log(\\"X Value from inner scope:\\", _testVariable);
console.log(\\"X Value from outer scope:\\", testVariable);
var source = {
  first: 1,
  second: 2
};
var destructed = Object.assign({
  third: 3
}, source);
console.log(\\"Destructed:\\", destructed);

var MyClass =
/*#__PURE__*/
function () {
  function MyClass() {
    console.log(\\"Called constructor\\");
    this.helper();
    this.instanceProperty = 3;
  }

  var _proto = MyClass.prototype;

  _proto.helper = function helper() {
    console.log(\\"Called helper\\");
  };

  return MyClass;
}();

var SecondClass =
/*#__PURE__*/
function (_MyClass) {
  _inheritsLoose(SecondClass, _MyClass);

  function SecondClass() {
    return _MyClass.call(this, 100) || this;
  }

  return SecondClass;
}(MyClass());

console.log(\\"Initialized class:\\", new MyClass());

function helper() {
  return new Promise(function ($return, $error) {
    return Promise.resolve(PromiseHelper()).then(function ($await_1) {
      try {
        return $return(42);
      } catch ($boundEx) {
        return $error($boundEx);
      }
    }, $error);
  });
}

helper();
//# sourceMappingURL=index.esm.js.map
"
`;

exports[`Publish Test File via Babel 3`] = `
"!function(e,o){\\"object\\"==typeof exports&&\\"undefined\\"!=typeof module?o(require(\\"@babel/runtime/helpers/inheritsLoose\\"),require(\\"core-js/modules/es6.object.assign\\"),require(\\"core-js/modules/web.dom.iterable\\"),require(\\"core-js/modules/es6.array.iterator\\"),require(\\"core-js/modules/es6.string.iterator\\"),require(\\"core-js/modules/es6.set\\"),require(\\"core-js/modules/es7.array.includes\\"),require(\\"core-js/modules/es6.promise\\"),require(\\"lodash\\")):\\"function\\"==typeof define&&define.amd?define([\\"@babel/runtime/helpers/inheritsLoose\\",\\"core-js/modules/es6.object.assign\\",\\"core-js/modules/web.dom.iterable\\",\\"core-js/modules/es6.array.iterator\\",\\"core-js/modules/es6.string.iterator\\",\\"core-js/modules/es6.set\\",\\"core-js/modules/es7.array.includes\\",\\"core-js/modules/es6.promise\\",\\"lodash\\"],o):o(e._inheritsLoose,null,null,null,null,null,null,null,e.lodash)}(this,function(e,o,r,s,l,n,t,i,u){\\"use strict\\"
e=e&&e.hasOwnProperty(\\"default\\")?e.default:e,console.log(\\"CherryPick Import Lodash:\\",\\"helloWorld\\"===u.camelCase(\\"hello world\\")),console.log(\\"Package\\",\\"preppy\\"),console.log(\\"Target\\",\\"lib\\"),console.log(\\"ES2016 Enabled:\\",4===Math.pow(2,2)),new Promise(function(e,o){e(\\"resolved\\")}).then(function(e){console.log(\\"Promise:\\",e)})
console.log(\\"Constant:\\",123)
var c=[1,2,3]
console.log(\\"Supports Array.includes?:\\",c.includes&&c.includes(2))
var a,d=new Set(c)
console.log(\\"Supports Set:\\",d.add(4)),void 0===a&&(a=!0),console.log(\\"Supports default parameters:\\",a)
console.log(\\"X Value from inner scope:\\",\\"inner\\"),console.log(\\"X Value from outer scope:\\",\\"outer\\")
var m=Object.assign({third:3},{first:1,second:2})
console.log(\\"Destructed:\\",m)
var p=function(){function MyClass(){console.log(\\"Called constructor\\"),this.helper(),this.instanceProperty=3}return MyClass.prototype.helper=function helper(){console.log(\\"Called helper\\")},MyClass}()
p()
console.log(\\"Initialized class:\\",new p),function helper(){return new Promise(function(e,o){return Promise.resolve(PromiseHelper()).then(function(r){try{return e(42)}catch(e){return o(e)}},o)})}()})
//# sourceMappingURL=index.umd.js.map
"
`;
